lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -16,532 +19,543 @@ $params = Parse-Args $args -supports_check_mode $true
+    if ($null -eq $choco_app) {
+        # We need to install chocolatey
+        # Enable TLS1.1/TLS1.2 if they're available but disabled (eg. .NET 4.5)
+        $security_protcols = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::SystemDefault
+        if ([Net.SecurityProtocolType].GetMember("Tls11").Count -gt 0) {
+            $security_protcols = $security_protcols -bor [Net.SecurityProtocolType]::Tls11
         }
-        elseif (-not $force)
-        {
-            return
+        if ([Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
+            $security_protcols = $security_protcols -bor [Net.SecurityProtocolType]::Tls12
         }
-    }
-
-    $options = @( "-y", $package, "--timeout", "$timeout", "--failonunfound" )
-
-    switch ($architecture) {
-      "x86" { $options += "--x86" ; break}
-    }
-
-    if ($check_mode)
-    {
-        $options += "--whatif"
-    }
-
-    if ($version)
-    {
-        $options += "--version", $version
-    }
+        [Net.ServicePointManager]::SecurityProtocol = $security_protcols
 
-    if ($source)
-    {
-        $options += "--source", $source
-    }
+        $client = New-Object -TypeName System.Net.WebClient
+        $environment = @{}
+        if ($proxy_url) {
+            # the env values are used in the install.ps1 script when getting
+            # external dependencies
+            $environment.chocolateyProxyLocation = $proxy_url
+            $web_proxy = New-Object -TypeName System.Net.WebProxy -ArgumentList $proxy_url, $true
+            $client.Proxy = $web_proxy
+            if ($proxy_username -and $proxy_password) {
+                $environment.chocolateyProxyUser = $proxy_username
+                $environment.chocolateyProxyPassword = $proxy_password
+                $sec_proxy_password = ConvertTo-SecureString -String $proxy_password -AsPlainText -Force
+                $web_proxy.Credentials = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $proxy_username, $sec_proxy_password
+            }
+        }
 
-    if ($force)
-    {
-        $options += "--force"
-    }
+        if ($source) {
+            # check if the URL already contains the path to install.ps1
+            if ($source.EndsWith("install.ps1")) {
+                $script_url = $source
+            } else {
+                # chocolatey server automatically serves a script at
+                # http://host/install.ps1, we rely on this behaviour when a
+                # user specifies the choco source URL. If a custom URL or file
+                # path is desired, they should use win_get_url/win_shell
+                # manually
+                # we need to strip the path off the URL and append install.ps1
+                $uri_info = [System.Uri]$source
+                $script_url = "$($uri_info.Scheme)/$($uri_info.Authority)/install.ps1"
+            }
+        } else {
+            $script_url = "https://chocolatey.org/install.ps1"
```

Comment:
Could we not have the `source` value default to 'https://chocolatey.org/install.ps1' ?
At least it would allow us to explicitly do: `source: null` if we don't want any installation.

Reply:
`source` needs to be `null` so that `choco.exe` will use the configured sources in it's config. If we were to set the default to be the public repo then user's won't be able to control their sources through `win_chocolatey_source`. Also the URL is a bit different from the Chocolatey install vs `choco.exe install`, the installer is looking for a script at a particular location while `choco.exe install` should have either a name or a source, URL to a nuget feed, or a path to a local directory containing sources.

Reply:
Oh right. Must have been confused.